### ISSUE TITLE
Nested attribute selector without & is parsed as raw

### ISSUE THREAD

**Issue by bartveneman** [csstree/csstree#268]
When parsing this (relaxed nesting) CSS we end up with a `Raw` node inside the first Rule, instead of another `Rule`:

```css
a {
  [test] {}
}
```

- Link to reproduction: https://stackblitz.com/edit/csstree-nested-attr-selector-repro?file=index.js
- CodePen showing that it actually works in a browser https://codepen.io/bartveneman/pen/XWOQLRz

When adding a `&` in front the parsing seems to work:

```css
a {
  & [test] {}
}
```

Maybe there's more relaxed nesting rules that don't work, but this is the only I've spotted so far.

**Comment by bartveneman** [comment]
Updated the above test case to csstree@3.0.0 and confirmed that the issue still exists.

**Comment by lahmatiy** [comment]
@bartveneman Yes, relaxed CSS nesting parsing is not supported yet, but it's a priority for implementation. Preparations have been made, but there are still some challenges to complete the implementation.

**Comment by bartveneman** [comment]
Excellent, looking forward to seeing it land. Thanks for confirming!

**Comment by nzakas** [comment]
https://github.com/csstree/csstree/pull/337
